### ISSUE TITLE
[bug] Libraries can contain characters that are not allowed in cmake target names

### ISSUE THREAD

**Issue by maddisoj** [conan-io/conan#9232]
### Environment Details (include every applicable attribute)
  * Operating System+version: Linux 5.12.11-arch1-1 x86_64
  * Conan version: 1.36.0
  * Python version: 3.9.5
  * CMake version: 3.20.4

### Steps to reproduce (Include if Applicable)
Install a conan package where one of its libraries has an `@` in the name with the `cmake_find_package` generator.
When calling `find_package()` for this package CMake will fail due to the `@`.

I believe this is the relevant docs from CMake: https://cmake.org/cmake/help/latest/policy/CMP0037.html
I might be able to work around it by removing those libs from `cpp_info.libs` and add a flag to `cpp_info.sharedlinkflags`.

### Logs (Executed commands with output) (Include/Attach if Applicable)
In my case I have created a conan package that exports the [Android VNDK](https://source.android.com/devices/architecture/vndk) from a build of AOSP. They have many, many libraries of the format: `<library>@<version>.so`.

CMake gives this error:
```
CMake Error at build_vndk/Findandroid-vndk.cmake:39 (add_library):
  The target name "CONAN_LIB::android-vndk_stagefright_bufferpool@2.0" is
  reserved or not valid for certain CMake features, such as generator
  expressions, and may result in undefined behavior.
Call Stack (most recent call first):
  build_vndk/Findandroid-vndk.cmake:125 (conan_package_library_targets)
  src/CMakeLists.txt:7 (find_package)


CMake Error at build_vndk/Findandroid-vndk.cmake:40 (set_target_properties):
  set_target_properties Can not find target to add properties to:
  CONAN_LIB::android-vndk_stagefright_bufferpool@2.0
Call Stack (most recent call first):
  build_vndk/Findandroid-vndk.cmake:125 (conan_package_library_targets)
  src/CMakeLists.txt:7 (find_package)
```

**Comment by memsharded** [comment]
Hi @maddisoj 

I am trying to write a test for this, and funny thing, is I cannot create a library with that name, because CMake directly will not allow it. Is it that VNDK is not using CMake to build? I guess I will have to build and then force a rename of the library...

**Comment by memsharded** [comment]
Proposing this fix for Conan 1.39: https://github.com/conan-io/conan/pull/9245

Note that the ``cmake_find_package`` generator will soon be deprecated in favor of ``CMakeDeps`` (which will be the only one that will remain in Conan 2.0). I have added the fix in both generators, but if for some reason this has some risk of breaking the ``cmake_find_package`` one, the fix might land only in ``CMakeDeps``.

**Comment by maddisoj** [comment]
Hi @memsharded 

AOSP uses its own build system, [soong](https://source.android.com/setup/build). I don't know of anything else that uses this.

Sadly, I can't share my VNDK recipe as it's a work project. Either way it exports binaries from AOSP, which takes an hour to clone, let alone build ;)

I appreciate that this is a weird edge case so thanks for fixing it :)

**Comment by memsharded** [comment]
Thanks for the info! :)

Lets see if this moves forward without problems. If you can't wait until the release:

- You can run from sources cloning my fork and checking out my branch, it is not difficult.
- As another, unrelated workaround, what I would have tried is to do a ``rename`` of the libraries in the ``package()`` method. It is perfectly fine to rename libraries, as long as the name is the correct one in ``self.cpp_info.libs``.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/9245 merged, this will be released in Conan 1.39
